libs/parallelism/execution/include/hpx/execution/sender_future.hpp

```diff
@@ -0,0 +1,61 @@
+#include <hpx/futures/future.hpp>
+#include <hpx/futures/promise.hpp>
+#include <hpx/type_support/pack.hpp>
+
+namespace hpx { namespace execution { namespace experimental {
+    namespace detail {
+        template <typename R>
+        struct make_future_receiver
+        {
+            hpx::lcos::local::promise<R> p;
+
+            void set_error(std::exception_ptr e) noexcept
+            {
+                p.set_exception(e);
+            }
+
+            void set_done() noexcept
+            {
+                std::terminate();
```

Comment:
Could this be called for `future<void>` under certain circumstances?

Reply:
The void-ness of the sender doesn't affect whether or not this can be called. This would be called if the sender is cancelled. We could set a special exception in that case (the terminate is mostly a placeholder right now) but it's not clear that's reasonable behaviour either.

In general I've done nothing so far to support cancellation for senders though so I might prefer to keep the terminate at least for the moment.

Reply:
Ok, makes sense.